src/controllers/UserController.js

```diff
@@ -181,4 +187,49 @@ export default class UserController {
       return resError(res, 500, error.message);
     }
   }
+
+  /**
+   * @description - this method Verifies a user
+   * @param {object} req - The request payload sent to the router
+   * @param {object} res - The response payload sent back from the controller
+   * @returns {object} - object
+   */
+  static async verifyEmail(req, res) {
+    try {
+      const userDetails = await Utils.decodeToken(req.query.token);
+      const user = await UserServices.getUserById(userDetails.id);
+      await UserServices.updateUserById({ name: 'isVerified', value: true }, user.id);
+      const homeLink = `${process.env.HOME_URL}`;
+      await NotifyUser.confirmUserVerified(user.email, homeLink);
+      return resSuccess(res, 200, {
+        message: 'Account verification was successful',
+      });
+    } catch (err) {
+      return resError(res, 500, 'Server error, Please try again later');
+    }
+  }
+
+  /**
+   * @description - this method Resends Verification Link to a user
+   * @param {object} req - The request payload sent to the router
+   * @param {object} res - The response payload sent back from the controller
+   * @returns {object} - object
+   */
+  static async resendVerifyEmail(req, res) {
+    try {
+      const { email } = req.body;
+      const data = await UserServices.getUserByEmail(email);
+      const { id } = data;
+      const user = await UserServices.getUserById(id);
```

Comment:
Why do you need to retrieve the user object here again from the database, doesn't the data variable assigned on line 221 hold the user object already.

Reply:
I retrieved it to send user email....When i used the data object, it did't send an email to user.

Reply:
What was the error you got, I believe the data object should hold the user's email as well which is basically what you need to send an email

Reply:
A send grid address error.... I guess when i did `data.email` and `data.firstName` it did't get both email and first name. so I got an error to specify the email address that i am sending the mail to.

So I thought its because i already destructured `id` to data that's why its not getting email and firstName from data object.